xc/common/primitives/ologice3/ologice3.pb_type.xml

```diff
@@ -32,24 +32,183 @@
   <output name="TQ" num_pins="1"/>
   <input name="SHIFTIN1" num_pins="1"/>
   <input name="SHIFTIN2" num_pins="1"/>
-  <mode name="PASS_THROUGH">
+
+  <!-- OLOGIC (ODDR/TDDR) or passthrough -->
+  <mode name="OLOGIC">
+
+    <!-- ODDR for T or passthrough -->
+    <pb_type name="OLOGIC_TFF" num_pb="1">
+      <input  name="TCE" num_pins="1"/>
+      <input  name="T1"  num_pins="1"/>
+      <input  name="T2"  num_pins="1"/>
+      <input  name="SR"  num_pins="1"/>
+      <input  name="CLK" num_pins="1"/>
+      <output name="TQ"  num_pins="1"/>
+      <output name="TFB" num_pins="1"/>
+
+      <mode name="PASSTHROUGH">
```

Comment:
Rather than model T1 -> TQ as a mode, couldn't this be modelled as a mux?  Because that is likely what is happening underneath, yes?

Reply:
So since Vivado actually does not allow to bypass T on an OSERDES then I should probably use a mux there. But I need to rethink it:

```
ERROR: [DRC THRU-1] OSERDES Route-thru Availability: Output buffer obuf_5 is driven by an OSERDES cell on the I pin and from fabric on the T pin. The current netlist does not have a legal routing. The OSERDES is used and no concurrent route-thru from fabric is supported. To resolve this violation, modify the design so that the output buffer pins are driven from the OSERDES (e.g. connect the OSERDES TQ port to the OBUF* T port).
```